### PR TITLE
fix: [CI-6003]: fixed error message in connection test

### PR DESCRIPTION
**Pull request by devkimittal**
## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- Immutable delegate `trigger publish-delegate`
</details>

<details>
  <summary>PR Check triggers</summary>

You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ti0, ti1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
    - CodeFormat: `trigger codeformat`
    - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- PMD: `trigger pmd`
- Feature Name Check: `trigger featurenamecheck`
- TI-bootstrap: `trigger ti0`
- TI-bootstrap1: `trigger ti1`
- TI-bootstrap2: `trigger ti2`
- TI-bootstrap3: `trigger ti3`
- TI-bootstrap4: `trigger ti4`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- Trigger all Checks: `trigger smartchecks`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/40229)

### CODE REVIEW

**Comment by devkimittal** [comment]
The error would look like this during connection test.
Screenshot 2022-11-16 at 3 16 40 PM

File: 932-connector-task/src/main/java/io/harness/connector/helper/DecryptionHelperViaManager.java [hunk 1]
```diff
@@ -25,6 +30,18 @@ public class DecryptionHelperViaManager implements DecryptionHelper {
...
+    checkForLocalEncryption(encryptionDetails);
     return restClientExecutor.getResponse(ngSecretDecryptionClient.decryptEncryptedDetails(build, "random"));
   }
+
+  private void checkForLocalEncryption(List<EncryptedDataDetail> encryptionDetails) {
+    for (EncryptedDataDetail encryptedDataDetail : encryptionDetails) {
+      EncryptedRecordData encryptedRecordData = encryptedDataDetail.getEncryptedData();
+      if (encryptedRecordData.getEncryptionType() != EncryptionType.LOCAL) {
+        throw new InvalidRequestException(format(
+            "While using connection via Harness Platform, Make sure your secrets are saved inside Harness Vault only.\nIn this case secret with identifier %s isn't saved inside Harness Vgithub_vault_patault",
```
Comment (rajatharanganath) [inline 932-connector-task/src/main/java/io/harness/connector/helper/DecryptionHelperViaManager.java:42]: Suggestion to change - Harness Vault > Harness Secret Manager
Also what is "Harness Vgithub_vault_patault" ?

Comment (rajatharanganath) [reply 932-connector-task/src/main/java/io/harness/connector/helper/DecryptionHelperViaManager.java:42]: Also check for spacing in 2 lines of the error's first character.

File: 932-connector-task/src/main/java/io/harness/connector/helper/DecryptionHelperViaManager.java [hunk 2]
```diff
@@ -25,6 +30,18 @@ public class DecryptionHelperViaManager implements DecryptionHelper {
   public DecryptableEntity decrypt(DecryptableEntity decryptableEntity, List<EncryptedDataDetail> encryptionDetails) {
     final DecryptableEntityWithEncryptionConsumers build =
         decryptableEntityHelper.buildDecryptableEntityWithEncryptionConsumers(decryptableEntity, encryptionDetails);
+    checkForLocalEncryption(encryptionDetails);
```
Comment (aman-harness) [inline 932-connector-task/src/main/java/io/harness/connector/helper/DecryptionHelperViaManager.java:33]: localEncryption is misleading name @devkimittal 
rename the fn to validateSecretManager.

File: 932-connector-task/src/main/java/io/harness/connector/helper/DecryptionHelperViaManager.java [hunk 3]
```diff
@@ -25,6 +30,18 @@ public class DecryptionHelperViaManager implements DecryptionHelper {
...
     return restClientExecutor.getResponse(ngSecretDecryptionClient.decryptEncryptedDetails(build, "random"));
   }
+
+  private void checkForLocalEncryption(List<EncryptedDataDetail> encryptionDetails) {
+    for (EncryptedDataDetail encryptedDataDetail : encryptionDetails) {
+      EncryptedRecordData encryptedRecordData = encryptedDataDetail.getEncryptedData();
+      if (encryptedRecordData.getEncryptionType() != EncryptionType.LOCAL) {
+        throw new InvalidRequestException(format(
+            "While using connection via Harness Platform, Make sure your secrets are saved inside Harness Secret Manager only.\nIn this case secret with identifier %s isn't saved in Harness Secret Manager",
```
Comment (aman-harness) [inline 932-connector-task/src/main/java/io/harness/connector/helper/DecryptionHelperViaManager.java:42]: Message is too verbose. We should not use \n in message. UI will take care of wrapping appropriately.
Connection via Harness Platform is not allowed if secrets are saved using custom secret manager. Review the secrets to use Connection via Harness Platform.

Comment (aman-harness) [reply 932-connector-task/src/main/java/io/harness/connector/helper/DecryptionHelperViaManager.java:42]: you can add secret lists as well.

**Comment by devkimittal** [comment]
trigger messagecheck

**Review by rajatharanganath — APPROVED** [review]

**Comment by devkimittal** [comment]
trigger messagecheck

File: 932-connector-task/src/main/java/io/harness/connector/helper/DecryptionHelperViaManager.java (hunk 3 above)
Comment (devkimittal) [reply 932-connector-task/src/main/java/io/harness/connector/helper/DecryptionHelperViaManager.java]: @aman-harness Made the change

**Review by aman-harness — APPROVED** [review]
:lgtm:

> Reviewed 1 of 1 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/40229#-:-NH3YRJT2zkZ8exLpN31:bnfp4nl)* status: 1. unresolved discussions.

**Review by devkimittal — COMMENTED** [review]
> Reviewed 1 of 1 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/40229)* status: 1. unresolved discussions.

**Review by rajatharanganath — APPROVED** [review]

**Review by rajatharanganath — APPROVED** [review]
> Reviewed 1 of 1 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/40229)* status: :shipit: complete! ✓

**Pull request merged**